sdk/src/cameras/3d-smart-camera/calibration_3d_smart.cpp

```diff
@@ -479,8 +479,8 @@ aditof::Status Calibration3D_Smart::distortionCorrection(uint16_t *frame,
             double y_dist_adim = y * m_distortion_cache[j * width + i];
 
             //back to original space
-            int x_dist = (int)(x_dist_adim * fx + cx);
-            int y_dist = (int)(y_dist_adim * fy + cy);
+            unsigned int x_dist = (unsigned int)(x_dist_adim * fx + cx);
```

Comment:
I wouldn't cast to unsigned here. If by any chance, the expression returns a negative value, casting it to an only positive value will populate x_dist and y_dist with undesired values. You can cast them inside the if (xdist >= 0 && x_dist < width &...)